src/views/search/components/AdvancedSearchModal/AdvancedSearchModal.tsx

```diff
@@ -0,0 +1,239 @@
+      dateFromString === '' &&
+      dateToString === '' &&
+      isNaN(vCPU.value) &&
+      isNaN(memory.value),
+    [
+      name,
+      projects,
+      description,
+      labels,
+      ip,
+      dateFromString,
+      dateToString,
+      vCPU.value,
+      memory.value,
+    ],
+  );
+
+  const onDateChange: (
+    setDateString: (dateString: string) => void,
+    setDate?: (date: Date) => void,
+  ) => DatePickerProps['onChange'] = (setDateString, setDate) => (_, dateString, date) => {
+    if (date) {
+      setDateString(dateString);
+      setDate?.(date);
+    } else {
+      setDateString('');
+      setDate?.(undefined);
+    }
+  };
+
+  const resetForm = () => {
+    setName('');
+    setProjects([]);
+    setDescription('');
+    setLabels([]);
+    setIP('');
+    setDateFromString('');
+    setDateToString('');
+    setVCPU(initialVCPU);
+    setMemory(initialMemory);
+    setIsValidDate(true);
+  };
+
+  const submitForm = () => {
+    onSubmit({
+      dateCreatedFrom: dateFromString,
+      dateCreatedTo: dateToString,
+      description,
+      ip,
+      labels,
+      memory,
+      name,
+      projects,
+      vCPU,
+    });
+  };
+
+  const closeModal = () => {
+    resetForm();
+    onClose();
+  };
+
+  return (
+    <Modal
+      className="ocs-modal co-catalog-page__overlay"
+      isOpen={isOpen}
+      onClose={closeModal}
+      position="top"
+      variant="medium"
+    >
+      <ModalHeader title={t('Advanced search')} />
+      <ModalBody>
+        <Form>
+          <FormGroup label={t('Name')}>
+            <TextInput onChange={(_, value) => setName(value)} type="text" value={name} />
+          </FormGroup>
+          <FormGroup label={t('Project')}>
+            <ProjectMultiSelect projects={projects} setProjects={setProjects} />
+          </FormGroup>
+          <FormGroup label={t('Description')}>
+            <TextInput
+              onChange={(_, value) => setDescription(value)}
+              type="text"
+              value={description}
+            />
+          </FormGroup>
+          <FormGroup label={t('Labels')}>
+            <LabelsMultiSelect labels={labels} setLabels={setLabels} />
+          </FormGroup>
+          <FormGroup label={t('IP')}>
+            <TextInput onChange={(_, value) => setIP(value)} type="text" value={ip} />
+          </FormGroup>
+          <FormGroup label={t('Date created')}>
+            <Split>
+              <DatePicker
+                onChange={onDateChange(setDateFromString, setDateFrom)}
+                placeholder={t('From')}
+                value={dateFromString}
+              />
+              <SplitItem className="pf-v6-u-pt-sm pf-v6-u-px-md">{t('to')}</SplitItem>
+              <DatePicker
+                validators={[
+                  (date) => {
+                    if (date < dateFrom) {
+                      setIsValidDate(false);
+                      return 'Date To cannot be before Date From';
+                    } else {
```

Comment:
no point in else here

Reply:
The `else` part was due to `return ''` to indicate there are no validation errors.